lib/std/fs.zig

```diff
@@ -501,7 +501,7 @@ pub const Dir = struct {
         },
         .linux => struct {
             dir: Dir,
-            buf: [8192]u8, // TODO align(@alignOf(os.dirent64)),
+            buf: [8192]u8 align(@alignOf(os.linux.dirent64)),
```

Comment:
I'm still a bit confused about this. Forcing the `buf` alignment is needed for the stage2 builder program to work on stricter arches like SPARC, but if I do this it seems that the code will fail to build on other OSes because they don't have a os.dirent64 struct.

Any workaround for this? Tried changing it to `os.linux.dirent64` but it doesn't seem to allow it to build on other OSes.

Reply:
> Any workaround for this?

`align(if (builtin.os.tag != .linux) 1 else @alignOf(os.dirent64))`

> Tried changing it to os.linux.dirent64 but it doesn't seem to allow it to build on other OSes.

If you follow all the `@import`s you'll notice how pretty much everything depends on `std.Target.current.os.tag`. When you're cross-compiling to eg. `i386-windows-gnu` the resulting `os.linux` module will contain some Linux definitions and some Windows ones coming from `bits/windows.zig` (!).

Reply:
Thanks, I've changed it following your direction.